core/src/martian/schema_tools.cljc

```diff
@@ -1,50 +1,370 @@
 (ns martian.schema-tools
-  (:require [schema.core :as s #?@(:cljs [:refer [MapEntry EqSchema]])]
-            [schema.spec.core :as spec])
-  #?(:clj (:import [schema.core MapEntry EqSchema])))
-
-;; todo
-;; write some tests and lean on schema-tools.core where possible
-
-(defn with-paths [path schema]
-  (keep (fn [schema]
-          (cond (and (instance? MapEntry schema)
-                     (instance? EqSchema (:key-schema schema)))
-                {:path (conj path (:v (:key-schema schema)))
-                 :schema (:val-schema schema)}
-                (map? schema)
-                {:path path
-                 :schema schema}))
-        (spec/subschemas (s/spec schema))))
+  (:require [camel-snake-kebab.core :refer [->kebab-case]]
+            [schema.core :as s]
+            [schema-tools.impl]))
+
+(defn explicit-key [k]
+  (if (s/specific-key? k) (s/explicit-schema-key k) k))
+
+(defn concrete-key?
+  "Checks if the schema key `k` is not generic (`s/Any`, `s/Keyword`, etc.)."
+  [k]
+  (or (keyword? k) (s/specific-key? k) (string? k)))
+
+(defn- can-be-renamed? [k]
+  ;; NB: See `camel-snake-kebab.internals.alter-name` ns.
+  (or (and (keyword? k) (not (namespace k))) (string? k)))
+
+(defn ->idiomatic [k]
+  (when-some [k' (explicit-key k)]
+    (when (can-be-renamed? k')
+      (->kebab-case k'))))
```

Comment:
A clearly readable intent of the function's logic, with reference to library-specific nuances.

The only logic changes are:
- untangling the condition for "idiomatizing" a key
- letting string keys to be "idiomatized" as well